sql3/planner/expression.go

```diff
@@ -2901,7 +2914,15 @@ func (p *ExecutionPlanner) compileCallExpr(expr *parser.Call) (_ types.PlanExpre
 		return agg, nil
 
 	default:
-		return newCallPlanExpression(parser.IdentName(expr.Name), args, expr.ResultDataType), nil
+		// could be a udf - try to look it up in functions
+		fn, err := p.getFunctionByName(strings.ToLower(callName))
+		if err != nil {
+			return nil, err
+		}
+		if fn != nil {
```

Comment:
So far as I can tell, this branch is irrelevant, because this cashes out to

```
if x != nil {
   return foo(x)
}
return foo(nil)
```

But that can actually just be done by `return foo(x)`. similarly, we can just call `newCallPlanExpression(..., fn)`, and if fn is nil, we'll do exactly what we're currently doing if fn is nil anyway, which is passing a nil in that slot.

Reply:
Yep - good catch